### DISCUSSION TITLE
Setting up new app using different ssh port

### DISCUSSION THREAD

**Discussion by Scrantch** [codemagic-ci-cd/codemagic-docs#1294] [Q&A] [answered]
Hi, I'm new to Codemagic and trying to setup our first app. Our Git server is self hosted (Gitea) in a Docker container and cannot use the stand SSH port. We need to use port 66. How do I do that? I tried using the standard Git URL format but it fails to connect. When I try on an external machine in shell it works just fine:

git@company.com:66/user/repository.git

How can I configure this to add our app to Codemagic?? Thanks!

**Comment by himesh-cm** [comment]
Hi @Scrantch,
Can you refer to this [documentation](https://docs.codemagic.io/getting-started/adding-apps/) on how you can add privately hosted repo.

**Reply by Scrantch** [reply]
Hi, yes I looked at that earlier. It says nothing about custom ports for SSH access.

**Reply by himesh-cm** [reply]
Can you share your team id?

**Reply by himesh-cm** [reply]
What error you get, when you try to connect?

**Reply by Scrantch** [reply]
I don’t have any of that yet. Literally at the signup setup screen for Codemagic trying to add the first app. When I put in the above URL repo format it says it can’t access it when it tries using git list remote. That leads me to believe it’s not actually looking at port 66.

**Reply by himesh-cm** [reply]
Okay, let me check. i will get back to you.

**Reply by himesh-cm** [reply]
Can you confirm if you have added generated public key to your repository settings?

**Reply by Scrantch** [reply]
Yep. Generated the key and put it in Git server.

**Reply by himesh-cm** [reply]
Thanks for confirming.

**Reply by Scrantch** [reply]
Any further ideas on this? Many thanks..

**Reply by himesh-cm** [reply]
@Scrantch, can you please try 
```
ssh://git@company.com:66/user/repository.git
```

**Reply by Scrantch** [reply]
Hi @himesh-cm as I stated in my post that’s what I initially tried. Doesn’t work. Is that the correct format? Is there an option somewhere to choose a different port? I assume this should be fairly trivial as many people don't allow traffic on default SSH port due to security reasons…

**Reply by Scrantch** [reply]
Here's what happens when I use that format. 

<img width="626" alt="Screenshot 2023-01-09 at 12 23 32 PM" src="https://user-images.githubusercontent.com/60194745/211401263-792daaf7-e4fe-48db-b860-0c45fca52ce9.png">

And yes, I have added the secure key to the Git server.

**Reply by himesh-cm** [reply]
Did you add `ssh://` as prefix?

**Reply by Scrantch** [reply]
Well damn. That was it! LOL The text in the format box didn't show the prefix. I assumed it figured it out. Working now. Thanks so much!

**Comment by himesh-cm** [comment, marked as answer]
The correct format its
`ssh://git@company.com:66/user/repository.git`
